### DISCUSSION TITLE
MNTOHA data exploration

### DISCUSSION THREAD

**Discussion by AndyMcAliley** [USGS-R/lake-temperature-lstm-static#12] [Show and tell]
## Exploring MNTOHA temperature observations

I looked at the availability of temperatures observations in the [MNTOHA dataset](https://www.sciencebase.gov/catalog/item/5e5c1b4fe4b01d50924f27e1). Check notebooks/Explore_MNTOHA_temperature_observations.ipynb or the associated html in this repo for the code that made these plots. This post is a wordier and code-less version of those files.

I wanted to get a sense for temperature observations are available for MNTOHA lakes. This affects the training of neural network models. Also, we might need to establish minimum data thresholds for including lakes in our training set.

# Temperature observations in each lake over time

First, I flipped through several lakes, looking at the distributions of observations in space (depth) and time. Run the notebook and keep rerunning the cell under the title "Plot temperature observations in each lake over time" to view observation distributions for lake after lake. Here are the first six:

![Turtle Lake](https://user-images.githubusercontent.com/11847289/154163641-2c1b2a4a-048a-4075-85d1-0cdec25f8c76.png)
![Big Stone Lake](https://user-images.githubusercontent.com/11847289/154163669-b9377dce-c89e-4e6a-b48a-aecfc753303b.png)
![Lac Qui Parle](https://user-images.githubusercontent.com/11847289/154163673-565a2a2b-0bea-487d-8976-fd3f0f3170af.png)
![Rice Lake](https://user-images.githubusercontent.com/11847289/154163678-29ed7454-c4ac-451d-b1da-752156506e45.png)
![Lake Traverse](https://user-images.githubusercontent.com/11847289/154163683-1dc52e08-7106-413e-a0f8-eb0b23ed0548.png)
![Buffalo Lake](https://user-images.githubusercontent.com/11847289/154163681-b77cc794-267f-4d21-bc44-e7c11ec66e43.png)

The black lines show the lake depth reported in `lake_metadata.csv`.

# Observations below nominal maximum lake depth

Oddly, many of these lakes have temperatures observed deeper than the nominal maximum depth. Strange. 526 out of 881 (877 with temperatures) have temperature observations deeper than their max depth! Here's the distribution of differences between nominal lake depth and maximum observed depth:
![lake_depth_vs_obs_depth](https://user-images.githubusercontent.com/11847289/154164258-ef8434e8-3136-4fc8-9d24-4e4bf2244389.png)
Positive values indicate lakes with reported depths that are deeper than the deepest temperature observation - what one might expect to be true of all lakes. Negative values are the surprising lakes that have temperature observations reported below their depth.
The distribution looks mostly symmetric, 0-centered, and heavy-tailed. I'd have expected that the observations deeper than the lake depth wouldn't be much deeper, and that there'd be more lakes without temperature observations near the lake bottom. I wonder how the lake depths in `lake_metadata.csv` were determined, and how it can be that so many observations are deeper than those reported depths?

# Lake observation thresholds

If we wanted to make sure that only "well-observed" lakes are included in training data, then we'd need to exclude lakes that don't have enough temperature observations. We could do that on the basis of the number of observation days (temporal), or the number of different observation depths (spatial), or a combination of both. For example, we could require that lakes have at least five days on which temperatures were observed for at least ten different depths.

![obs_available](https://user-images.githubusercontent.com/11847289/154166716-41be3c4a-536c-4b51-b1da-f217dcc1e446.png)
The labeled contours and the color indicate how many lakes meet combinations of temporal and spatial data thresholds. Careful: the x-axis, the colors, and the contours are spaced logarithmically. The y-axis is linear. I chose to do this because it allows a wide range of data thresholds to be illustrated on one figure.

# Occurrence of temperature values in dataset

Sometimes artifacts of the data collection process may appear in a dataset as repeated values. For example, a faulty sensor might always report the same value. If they are not recognized, such values can corrupt the data and distort models and inferences based on those data. Let's check for such repeated values in this dataset by plotting the most common temperature values.

![common_obs_temperatures](https://user-images.githubusercontent.com/11847289/154166754-e3f7e27d-cfcd-47dc-bf1f-9be9436769ed.png)

Round numbers greater than 4&deg;C are most common, then values rounded to the nearest 0.5&deg;C. So, some of the data were likely rounded to the nearest degree, or half degree. The prevalence of 20&deg;C indicates that some temperatures might have been rounded to the nearest five or ten degrees. It's reassuring to see that 0&deg;C isn't too high in the list. It's interesting that 5&deg;C is second-most common. Maybe that's the most common temperature of the hypolimnion?

Here's a plot of common temperature values collected after the year 2000:
![common_obs_temperatures_2000s](https://user-images.githubusercontent.com/11847289/154169636-54b554bc-6855-4c72-85f3-0e038734af19.png)
5&deg;C is nowhere to be seen. Looks like that value's prevalence in the data is an artifact of older data collection or recording techniques and practices. 20&deg;C still tops the chart, though.

**Comment by jordansread** [comment]
Andy - I'm honing in on the depth QAQC part of this, because I think it is am important thing to keep an eye on. We have a supporting issue [here](https://github.com/USGS-R/lake-temperature-model-prep/issues/227) which I think needs attention from our team. There are clearly some errant depths in the dataset, but there are also a few explanations for why we'd see deeper temperature obs than max depth in other cases

First, lakes can swing by ~2meters between the wettest periods and the driest periods. This swing is less if the lake is a flowage lake vs a "drainage" or "perched" lake. And the highest swings are on reservoirs which can be tens of meters. So sometimes the max depth is based on a period where the lake wasn't at its deepest. 

Second, sometimes the max depth comes from one basin but it is applied to the whole lake. We've seen this in situations where the state DNR splits a lake into two (or three) and we/USGS/NHDHR considers it to be one lake. So if we only have a max depth from one of those three basins/lakes, our pipeline would apply it to the whole NHDHR lake, and sometimes that means that isn't the deepest part of the lake (we've sniffed this out for at least one lake in MI and it is likely in MN too). 

Lastly, there are reporting errors in the temperatures and their associated depths. This is part of the QA issues we have with the temperature data. 

Hope this helps!
